pulser/sequence.py

```diff
@@ -834,13 +870,23 @@ def align(self, *channels: Union[str, Parametrized]) -> None:
         if self.is_parametrized():
             return
 
-        last_ts = {id: self._last(cast(str, id)).tf for id in channels}
+        channels = cast(Tuple[str], channels)
+        last_ts = {
+            id: self.get_duration(id, include_fall_time=True)
+            for id in channels
+        }
         tf = max(last_ts.values())
 
         for id in channels:
             delta = tf - last_ts[id]
```

Comment:
That's already being done a little further down

Reply:
From my understanding, it is done for each channel separately, but they might not all have the same `min_duration`. Thus, in my sense, they should all align to the highest `min_duration`. I think my suggestion is slightly different from what there is further down because it applies the max of `min_duration` of _all channels_.

Reply:
Maybe it doesn't matter actually, if the delays on the channels don't need to all finish at the same time, and if the only thing that matters is that each channel will finish after all instructions of other channels.

Reply:
Yes, that would be enough. Plus, the channels will usually have the same `min_duration` so that's not a big concern anyway.

Reply:
Also, note that generally that level of alignment is not guaranteed because the channels maybe have different clock periods (which is, again, not a likely scenario).